File: backend/src/main/scala/order/api/SendOrderChatMessageApi.scala
package order.api

import domain.shared.given

import admin.app.*
import cats.effect.IO
import domain.order.SendOrderChatMessageRequest
import domain.shared.{ApprovalFlag, CustomerId, DeliveryAppState, OrderId, UserRole}
import org.http4s.HttpRoutes
import org.http4s.circe.CirceEntityCodec.*
import org.http4s.dsl.io.*
import shared.api.routing.*
import shared.app.*

val sendOrderChatMessageApi: FixedMethodApi1[OrderId, DeliveryAppState] =
  jsonPostApi1[OrderId, SendOrderChatMessageRequest, DeliveryAppState](
    List(routeSegment("api"), routeSegment("delivery"), routeSegment("orders")),
    List(routeSegment("chat")),
  )

File: frontend/src/apis/order/SendOrderChatMessageApi.ts
import type { DeliveryAppState, OrderId, SendOrderChatMessageRequest } from '@/objects/core/SharedObjects'
import { postNormalizedDeliveryState } from '@/system/api/DeliveryStateClient'
import { buildApiPath1, defineJsonPostApi1, routeSegment } from '@/system/api/TypedApiDefinitions'

export const sendOrderChatMessageApiDefinition = defineJsonPostApi1<OrderId, SendOrderChatMessageRequest, DeliveryAppState>(
    [routeSegment('api'), routeSegment('delivery'), routeSegment('orders')],
    [routeSegment('chat')],
  )

export function sendOrderChatMessage(
  orderId: OrderId,
  payload: SendOrderChatMessageRequest,
) {
  return postNormalizedDeliveryState(buildApiPath1(sendOrderChatMessageApiDefinition, orderId), payload)
}
